### ISSUE TITLE
Timeout dialog doesn't read out heading in VoiceOver

### ISSUE THREAD

**Issue by edwardhorsford** [hmrc/hmrc-frontend#387]
I'm doing some quick ad-hoc testing of the timeout dialog in Voiceover on Safari on Mac and iOS.

In both, when the dialog is shown, the paragraph gets read out, but not the heading. My expectation is that the heading should get read out too - at least on first load.
![Screenshot 2024-07-08 at 15 59 37](https://github.com/hmrc/hmrc-frontend/assets/2204224/4872390f-1d01-497e-8d6e-590b6ad14696)

Possibly relevant, I note the paragraph gets `aria-live="assertive"` but the h1 does not.

**Comment by shabana-ali** [comment]
Hi @edwardhorsford 

I just did a quick test with VoiceOver and Safari (Version 17.5). On load it read out "You’re about to be signed out". Is that what is not being announced for you?

https://github.com/hmrc/hmrc-frontend/assets/59563139/58c48fd3-a5e5-47e0-a262-14365bbdf1ad

Apologies - not had chance to test with iOS.

My understanding for the `aria-live="assertive"` on the paragraph is to announce the count down when it changes (at 20 second increments).

**Comment by edwardhorsford** [comment]
@shabana-ali For voiceover on Mac and iOS I have not heard it read out the h1.

**Comment by edwardhorsford** [comment]
@shabana-ali Here's a recording of VoiceOver on iOS. The mac was similar.


https://github.com/hmrc/hmrc-frontend/assets/2204224/02380eda-669b-4cd5-ba60-8b4527499d15

**Comment by edwardhorsford** [comment]
I've tried various combinations of `aria-live="polite"` and `aria-live="assertive"` on the heading, and it never gets read out on iOS if the second paragraph has any form of `aria-live`. I might suspect this could be a bug with VoiceOver on iOS and multiple aria-live regions.

**Comment by edwardhorsford** [comment]
[Thread here](https://github.com/w3c/aria/issues/1689#issuecomment-1026892454) on issues with multiple live regions, including this line:

> BTW I have also experienced the VO issue where live regions upstage each other completely. 

I think this isn't a major bug though - as the paragraph in isolation already gives a clear message about what is going on.

A possible solution would be to make the heading hidden from screen readers, and include the existing heading in the audible message the first time it gets rendered.

**Comment by ashfaqhussain357** [comment]
Hi @edwardhorsford, is this still an issue? The last time we spoke you mentioned you were going to test this with a combination screen readers.

**Comment by edwardhorsford** [comment]
Hi @ashfaqhussain357 yes I think so. It's not a huge issue as a meaningful message is still read out, but the first heading gets missed.

I suspect the solution is to refactor how these messages work. From my research, I think there should only be a single live region that gets updated. So I feel like a refactor could have a single live area that initially gets the heading and paragraph inserted in to it, then later just the paragraph.

**Comment by oscarduignan** [comment]
I don't know if anything has changed recently (and it has been a while between, so apologies it's taken so long to get to working on this fix) but we were just testing this out and we can't recreate the bug at the moment, video attached

We will still go ahead with our planned fix though, where we will initially create the audible message without `aria-live="assertive"` and only add `aria-live="assertive"` when it's updated

At the end of the video you'll see I try tabbing and it doesn't let me tab to the sign out link, I thought this might be a bug, but in further testing I remembered that safari changes how tabbing works by default, so I think that's it. Where I have recently changed my computer, and haven't yet unticked the setting to not mess with how tabbing works 👍 just disclaiming it for anyone watching the recording who, like me, had forgotten about this because they don't often use safari

https://github.com/user-attachments/assets/a2e6c01d-fdc6-4f57-81cf-5899e54b8e47

**Comment by oscarduignan** [comment]
We tried leaving out the aria-live="assertive" on first render, but testing that, it seems like then it doesn't announce the updates. I don't have a very deep understanding of aria-live, so maybe it's correct that it needs to be present on the element when the element is added to the page for future changes to be announced, or maybe there's something else going on. Two recordings, first without devtools open, second with devtools open showing how the message is updated

https://github.com/user-attachments/assets/0be63de1-23da-407c-81c1-cc3755551d3d

https://github.com/user-attachments/assets/b837f7c4-a959-4df7-84c1-69eb0ddb46db

**Comment by shabana-ali** [comment]
We have now carried out some tests with the current [service timeout pattern](https://design.tax.service.gov.uk/hmrc-design-patterns/service-timeout/) and where the `aria-live="assertive"` was placed on the container `div` (as opposed to on the hidden copy). We tested with the following screen readers:
- VoiceOver with Safari and Chrome on MacOS
- JAWS with Chrome on Windows
- NVDA with Chrome on Windows
- Voiceover with Safari on iOS
- TalkBack with Chrome on Android

With the current service timeout, Voiceover on iOS was the only one that did not announce the "You’re about to be signed out" heading. We felt there were not any other issues with this implementation.

With the updated test version, even though the heading was announced with Voiceover on iOS, we felt this produced a poorer and nosier experience with some of the other screen readers; the countdown was not announced with VoiceOver with Chrome (although this is not a common screen reader browser combination) and on the mobile devices the button/link text was read out without relying their role.

We have therefore decided not to update the current service timeout implementation (the code is technically correct for what we want to achieve). The countdown announced by Voiceover on iOS does sufficiently convey what is happening and should not cause the user any confusion.

@edwardhorsford Thanks for raising this and I hope this outcome is acceptable.

We will be happy to relook at this again in the future if deemed problematic.
